src/services/WebViews.ts

```diff
@@ -192,15 +230,52 @@ export class WebViews {
 		} else {
 			fullSourcePath = this.getFullSourcePathIfExistsForBoundProject(node.FileName[0]);
 		}
+
 		const uri = vscode.Uri.file(fullSourcePath);
 		let found: boolean = this.isTextEditorVisible(node, fullSourcePath);
 		if (!found) {
 			vscode.commands.executeCommand('vscode.open', uri, vscode.ViewColumn.One).then(() => {
 				found = this.isTextEditorVisible(node, fullSourcePath);
 			});
 		}
+		
+	}
+	private async resultStateChanged(selectedResultState: any, rows: any) {
+		let scanId= this.scanNode.scanId
+		let nodes = this.queryNode.Result;
+		//The below for loop updates the result state
+		for (var i = 0; i < rows.length; i++) {
+			var pathId = rows[i];
+			for (let i = 0; i < nodes.length; i++) { 
```

Comment:
Can you please check if it is working properly because I cann see in outer loop as well as in inner loop i variable is used